dev/knowledge/backend/architecture.md

```diff
@@ -0,0 +1,61 @@
+# Backend Architecture
+
+> Part of: `dev/knowledge/backend/` | Related: [AGENTS.md](../../../backend/AGENTS.md)
+
+High-level overview of the Infrahub backend architecture.
+
+## Tech Stack
+
+- **Framework:** FastAPI 0.121.1 (async-first)
+- **Database:** Neo4j 5.28 (graph database)
+- **API:** GraphQL (primary) + REST
+- **Python:** 3.12 with Pydantic 2.10
```

Comment:
Should we remove references to versions? It's already outdated since you target `stable`